client/state/data-layer/wpcom/sites/activity/schema.js

```diff
@@ -0,0 +1,174 @@
+			properties: {
+				external_user_id: { type: 'integer' },
+				icon: {
+					type: 'object',
+					required: [ 'url' ],
+					properties: {
+						url: { type: 'string' },
+						width: { type: 'integer' },
+						height: { type: 'integer' },
+					},
+				},
+				name: { type: 'string' },
+				role: { type: 'string' },
+				type: { type: 'string' },
+				wpcom_user_id: { type: 'integer' },
+			},
+		},
+		generator: { type: 'object' },
+		gridicon: { type: 'string' },
+		items: {
+			type: 'array',
+			items: { type: 'object' },
+		},
+		name: {
+			type: 'string',
+			// eslint-disable-next-line quote-props
+			enum: [
+				'attachment__deleted',
+				'attachment__updated',
+				'attachment__uploaded',
+				'comment__approved',
+				'comment__content_modified',
+				'comment__deleted',
+				'comment__published',
+				'comment__published_awaiting_approval',
+				'comment__spammed',
+				'comment__trashed',
+				'comment__unapproved',
+				'core__autoupdated',
+				'core__network_updated',
+				'core__reinstalled',
+				'core__update_available',
+				'core__updated',
+				'feedback__published',
+				'jetpack__site_connected',
+				'jetpack__site_disconnected',
+				'jetpack__user_linked',
+				'jetpack__user_unlinked',
+				'menu__added',
+				'menu__deleted',
+				'menu__updated',
+				'plugin__activated',
+				'plugin__autoupdated',
+				'plugin__deactivated',
+				'plugin__deleted',
+				'plugin__deletion_failed',
+				'plugin__edited',
+				'plugin__installed',
+				'plugin__installed_filesystem',
+				'plugin__network_activated',
+				'plugin__network_deactivated',
+				'plugin__update_available',
+				'plugin__updated',
+				'post__deleted',
+				'post__exported',
+				'post__imported',
+				'post__publicized',
+				'post__published',
+				'post__trashed',
+				'post__updated',
+				'rewind__complete',
+				'rewind__error',
+				'term__created',
+				'term__deleted',
+				'term__edited',
+				'theme__deleted',
+				'theme__edited',
+				'theme__installed',
+				'theme__network_disabled',
+				'theme__network_enabled',
+				'theme__switched',
+				'theme__update_available',
+				'theme__updated',
+				'user__added',
+				'user__deleted',
+				'user__deleted-reassigned',
+				'user__failed_login_attempt',
+				'user__login',
+				'user__logout',
+				'user__registered',
+				'user__removed',
+				'user__updated',
+				'widget__added',
+				'widget__edited',
+				'widget__inactive',
+				'widget__inactive-cleared',
+				'widget__removed',
+				'widget__reordered',
+			],
+		},
+		object: { type: 'object' },
+		published: { type: 'string' },
+		status: {
+			oneOf: [
+				{ type: 'null' },
+				{
+					type: 'string',
+					// eslint-disable-next-line quote-props
+					enum: [ 'error', 'info', 'success', 'warning' ],
+				},
+			],
+		},
+		summary: { type: 'string' },
+		target: { type: 'object' },
+		totalItems: {
+			min: 0,
+			type: 'integer',
+		},
+		type: { type: 'string' },
+	},
+	additionalProperties: false,
+};
```

Comment:
this is an unanswered question, but it would be nice to start thinking about how to turn these into a reusable and sharable library. having them in a flat directory could help. storing them as JSON could help (for PHP)

just earmarking